python/dcr_template.py

```diff
@@ -919,36 +1014,47 @@ def build_model(self, use_full=True, use_regularization=True, use_only_detected=
                 if use_only_detected:
                     if self.mask[j, i] & detected_bit == 0:
                         continue
-                img_vals = self._extract_image_vals(j, i, radius=pix_radius)
-
-                model_vals = self.solve_model(dcr_kernel, img_vals, use_regularization=use_regularization)
-                self._insert_model_vals(j, i, model_vals, radius=pix_radius)
+                img_vals = self._extract_image_vals(j, i, image_arr, mask=mask_arr, radius=pix_radius)
+
+                model_vals = self.solve_model(self.kernel_size, self.n_step, kernel_base, img_vals,
+                                              lstsq_kernel, use_nonnegative=use_nonnegative,
+                                              use_regularization=use_regularization,
+                                              regularization=self.regularize)
+                self._insert_model_vals(j, i, model_vals, model, weights,
+                                        radius=pix_radius, kernel=self.psf_avg)
         if verbose:
             print("\nFinished building model.")
+        self.weights = weights
+        self.model = [model[f, :, :] for f in range(self.n_step)]
 
-    def build_dcr_kernel(self, use_full=None):
-        """!Calculate the DCR covariance matrix for a set of exposures.
+    @staticmethod
+    def build_lstsq_kernel(dcr_kernel, regularization=None, use_regularization=True, kernel_weight=None):
+        """!Build the matrix of the form M = (A^T A)^-1 A^T for a linear least squares solution.
 
-        @param use_full  Flag, set to True to use measured PSF for each exposure,
-                            or False to use the fiducial psf for each.
+        @param dcr_kernel  The covariance matrix describing the effect of DCR
+        @param regularization  Regularization matrix created with build_regularization
+        @param use_regularization  Flag, set to True to use regularization. The type of regularization is set
```

Comment:
Yes, I'll do that. In a previous version, regularization was calculated early on and it was important to be able to switch between using it or not, but that feature is now obsolete.